playbooks/files/rax-maas/plugins/ironic_capacity_check.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+
+# Copyright 2017, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+
+import ipaddr
+from ironicclient import exc
+from maas_common import get_auth_ref
+from maas_common import get_ironic_client
+from maas_common import metric
+from maas_common import metric_bool
+from maas_common import print_output
+from maas_common import status_err
+from maas_common import status_ok
+
+
+def check(auth_ref, args):
+
+    IRONIC_ENDPOINT = ('http://{ip}:6385/v1'.format(ip=args.ip))
+
+    try:
+        if args.ip:
+            ironic = get_ironic_client(endpoint=IRONIC_ENDPOINT)
+        else:
+            ironic = get_ironic_client()
+
+        is_up = True
+
+    except exc.ClientException:
+        is_up = False
+    # Any other exception presumably isn't an API error
+    except Exception as e:
+        metric_bool('client_success', False, m_name='maas_ironic')
+        status_err(str(e), m_name='maas_ironic')
+        return
+    else:
+        metric_bool('client_success', True, m_name='maas_ironic')
+        # pass limit=0 to list all nodes list without pagination
+        all_nodes = ironic.node.list(limit=0)
+        status_ok(m_name='maas_ironic')
+
+    if is_up:
+        maint_nodes = [node for node in all_nodes if node.maintenance]
+        maint_nodes_count = len(maint_nodes)
+        total_nodes = len(all_nodes)
+        up_nodes = total_nodes - maint_nodes_count
+        metric('ironic_maint_nodes_count', 'uint32', maint_nodes_count)
```

Comment:
It looks like the check is only using `up_nodes` and `total_nodes`—do we need to be sending `maint_nodes_count` here as well?